Latejoin QOL: Interview holograms

whatston3 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

Part 2 of latejoin QOL efforts.

Latejoining onto ships now starts you off as a hologram.  Both the hologram and the captain of the ship need to approve the process in order to actually join the ship.  Anyone with a deed to the ship is considered an acting captain, and you _must currently have the ID on you_.

If the captain or applicant wish to leave, they can at any time.  The captain interacts with the hologram through right click (ideally left click brings up a rotary menu).

Holograms cannot interact with the world, though they aren't ghosts - they take the (approximate) physical appearance of their character.  Ghosting from a hologram should delete it and reopen the job slot, the hologram should not be able to leave the ship or meaningfully interact with the world.

The feature is currently missing the ability to prevent a user by ID from rejoining the crew.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

Latejoin QOL and usability.

## Technical details
<!-- Summary of code changes for easier review. -->

New system, new component for tracking interview hologram entities, and new entities for hologram types.  There are new jobs (Contractor/Pilot/Merc Applicant) that start out as these holograms, and all shuttles are swapped over to use these new job slots.

## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->

1. Start two clients.  On client 2, ensure your character has merc/pilot/contractor loadouts with a known cost.
2. Buy a ship on client 1, move over to the ship, open two contractor slots.  Make sure your ID is in your PDA, which is in your ID slot.
3. Latejoin the ship on client 2, you should start as a hologram with two actions.
4. On client 1, the captain's PDA should ring and you should get a notification (you should hear it on both clients).
5. Press the paper on fire action, you should return to the lobby and have a notification that you left in the chat.
7. Eject the ID from client 1's PDA.
8. On client 2, rejoin the ship (there should still be two slots open).
9. Press the Approve button.
10. On client 1, right click the hologram, "Approve" should be disabled.
11. Pick up the captain's ID.
12. Right click the hologram again and press "Approve".
13. The character should be replaced with their actual character.
14. Check the station records, your character details should show up and be correct.
15. Check the criminal records, your character details should show up and be correct.
16. Respawn on client 2.  Your loadout cost should be deducted from your funds.
17. Rejoin as an applicant (there should be one contractor slot left open).
18. On client 1, right click on the hologram, press "dismiss".  Client 2 should exit back to the lobby, have a message telling them they were dismissed, and the slot should remain open.
19. On client 2, rejoin as a Contractor Applicant, should still be open with one slot.
20. On client 1, right click on the hologram, select "Dismiss and Close".  Client 2 should exit back to the lobby, have a message telling them they were dismissed, and the slot should be closed.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

A slime hologram applies for a contractor job on a Pioneer.  Note the actions in the top left of the screen.  The captain of the ship is present for the interview.
![image](https://github.com/user-attachments/assets/9613b4fa-44a4-4247-a49e-3154df3b4509)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

Ship submissions should swap over to the ContractorInterview, PilotInterview, and MercenaryInterview jobs.

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- add: Most latejoin slots on shuttles must interview as holograms before joining.  See the Hiring Crew->Interviews guidebook page for details.

dvir001 commented:
I joked about adding LinkedIn, but he went and did it anyway.

MagnusCrowe commented:
Seems to work.  I would have no problem putting this in as is. 

I would recommend the following be implemented either in the future or in this PR if you're feeling saucy. 

Could kill off the hologram much earlier when they leave the ship.
Could give the PDA that contains the captain's ID a buzz/beep and a message about an applicant.
Could give the applicant a server message when they are forced back to lobby to explain why (maybe they were outside the ship too long, maybe they were denied, maybe they revoked their application).
Could restore the open slot when the applicant revokes their app, dies, or the captain denies them.
Could put a timer on the applicant to stop them from reapplying to a ship they were denied from (to prevent annoying the captain because some salty baby got denied).
Could let holograms phase through doors instead of opening them (like mice).

whatston3 commented:
> Could kill off the hologram much earlier when they leave the ship.

Cut it down by 5 seconds to 10.

> Could give the PDA that contains the captain's ID a buzz/beep and a message about an applicant.

Maybe.  I'll see what I can do.

> Could give the applicant a server message when they are forced back to lobby to explain why (maybe they were outside the ship too long, maybe they were denied, maybe they revoked their application).

Yeah, that'd be nice.  Outside of the ship might be tricky, it's just a timed despawn at the moment.

> Could restore the open slot when the applicant revokes their app, dies, or the captain denies them.

Fixed, unintentional.  Fixed a few issues with ghosting as well.

> Could put a timer on the applicant to stop them from reapplying to a ship they were denied from (to prevent annoying the captain because some salty baby got denied).

Larger problem, but sure.  I think a block command might be the best option, but even that requires a bit more work.  Simplest option could be a "dismiss without reopening the slot" and locking down your slot for a while.  Maybe if they could be banished by hitting them?  Dunno.

> Could let holograms phase through doors instead of opening them (like mice).

I changed their layer, mask, and removed the door bump opening (not that it really matters anymore) - it's a bit odd because they'll show up on top of tables (they're still on the mob draw layer, and that's probably for the best).

Fixed a few issues with odd icons in the verb menu, and now options show up regardless of whether or not you're a captain.  The holograms are slower, and can be dispelled as long as you have line of sight on them.  Worst case, they're about as annoying as a mouse, they can talk, but can't interact with most things.

Houtblokje commented:
> Simplest option could be a "dismiss without reopening the slot" and locking down your slot for a while. 

That would punish the captain for not wanting a candidate though, since they won't be able to get a new candidate for a bit.

whatston3 commented:
Functionally, I think this is alright.

When somebody joins an interview slot, anyone with a PDA with the deed to the shuttle will get a notification as shown below.  All fields are localized.

![image](https://github.com/user-attachments/assets/4c7c31ba-7e6d-45d8-abd6-904688fcf85e)

The applicant gets a record in the ship they join, along with a criminal database record.  Note that both the jobs are Contractor, not Contractor Applicant.
![image](https://github.com/user-attachments/assets/2db99d1e-3c7b-4be5-928b-d7ce2c773e2f)
![image](https://github.com/user-attachments/assets/f131fd52-99c5-4aee-833d-ccd5d9463582)

When the interview ends in cancellation, dismissal, or approval, the applicant is notified.
![image](https://github.com/user-attachments/assets/b5312676-0e1d-451b-adac-424ae85faf4e)
![image](https://github.com/user-attachments/assets/fe3cf06a-419e-41f3-8bc9-103eafc05a15)
![image](https://github.com/user-attachments/assets/925e9fbc-7a18-4a46-b93b-bc1603d0e0e4)

As a means to mitigate griefing, the option to dismiss someone and close their job slot was added.
![image](https://github.com/user-attachments/assets/0da50311-3f83-4f3a-9627-1f8f97c3a7a6)

Fixed an issue where portable generators could be turned off by holograms.
